addons/website_crm_partner_assign/static/src/js/crm_partner_assign.js

```diff
@@ -1,202 +1,256 @@
 odoo.define('crm.partner_assign', function (require) {
 'use strict';
 
+var sAnimations = require('website.content.snippets.animation');
 var rpc = require('web.rpc');
 var weContext = require('web_editor.context');
-require('web.dom_ready');
 /*
  * This file is intended to add interactivity to survey forms rendered by
  * the website engine.
  */
 
-var interested_form = $('.interested_partner_assign_form');
-var desinterested_form = $('.desinterested_partner_assign_form');
-var opp_stage_buttons = $('.opp-stage-button');
-var new_opp_form = $('.new_opp_form');
+sAnimations.registry.crmPartnerAssign = sAnimations.Class.extend({
+    selector: '#portal_my_lead',
+    read_events: {
+        'click .interested_partner_assign_confirm': '_onInterestedPartnerAssignConfirm',
+        'click .desinterested_partner_assign_confirm': '_onDesinterestedPartnerAssignConfirm',
+        'click .opp-stage-button': '_onOppStageButtons',
+        'change .edit_contact_form .country_id': '_onEditContactForm',
+        'click .edit_contact_confirm': '_onEditContactConfirm',
+        'click .new_opp_confirm': '_onNewOppConfirm',
+        'click .edit_opp_confirm': '_onEditOppConfirm',
+        'change .edit_opp_form .next_activity': '_onEditOppConfirm',
+        'click div.input-group span.fa-calendar': '_onCalendarIcon'
+    },
 
-if(!interested_form.length && !desinterested_form.length && !opp_stage_buttons.length && !new_opp_form.length) {
-    return $.Deferred().reject("DOM doesn't contain website_crm_partner_assign elements");
-}
+    /**
+     * @override
+     * @param {Object} parent
+     */
+    start: function (parent) {
+        var interested_form = $('.interested_partner_assign_form');
+        var desinterested_form = $('.desinterested_partner_assign_form');
+        var opp_stage_buttons = $('.opp-stage-button');
+        var new_opp_form = $('.new_opp_form');
 
-$('.interested_partner_assign_confirm').on('click',function(e){
-    var $btn = $(this);
-    if ($('.interested_partner_assign_form .comment_interested').val() && $('.interested_partner_assign_form .contacted_interested').prop('checked')){
+        if (!interested_form.length && !desinterested_form.length && !opp_stage_buttons.length && !new_opp_form.length) {
+            return $.Deferred().reject("DOM doesn't contain website_crm_partner_assign elements");
+        }
```

Comment:
This would do nothing. Previous code checked that those elements were in the DOM and if not, did nothing. With your new code, the animation is started, then the start method is rejected (which does nothing) and all the handlers are made active anyway. What you have to do here is define an animation selector with something like:
```javascript
selector: '#portal_my_lead:has(.interested...)'
```
That way the animation will always be defined but only instantiated if it finds a portal_my_lead which *has* the necessary stuff. Not sure if I am clear ?